tests/Umbraco.Tests.Integration/Umbraco.Infrastructure/Persistence/Repositories/DocumentRepositoryTest.cs

```diff
@@ -179,40 +179,124 @@ public void CacheActiveForIntsAndGuids()
 
             udb.EnableSqlCount = false;
 
-            var template = TemplateBuilder.CreateTextPageTemplate();
-            FileService.SaveTemplate(template);
-            var contentType =
-                ContentTypeBuilder.CreateSimpleContentType("umbTextpage1", "Textpage", defaultTemplateId: template.Id);
-
-            contentTypeRepository.Save(contentType);
-            var content = ContentBuilder.CreateSimpleContent(contentType);
-            repository.Save(content);
+            var content = CreateContent(repository, contentTypeRepository);
 
             udb.EnableSqlCount = true;
 
-            // go get it, this should already be cached since the default repository key is the INT
+            // Initial and subsequent requests should use the cache, since the cache by Id and Key was populated on save.
             repository.Get(content.Id);
             Assert.AreEqual(0, udb.SqlCount);
 
-            // retrieve again, this should use cache
             repository.Get(content.Id);
             Assert.AreEqual(0, udb.SqlCount);
 
-            // reset counter
+            repository.Get(content.Key);
+            Assert.AreEqual(0, udb.SqlCount);
+
+            repository.Get(content.Key);
+            Assert.AreEqual(0, udb.SqlCount);
+        }
+    }
+
+    [Test]
+    public void Retrieval_By_Key_After_Retrieval_By_Id_Is_Cached()
+    {
+        var realCache = new AppCaches(
+            new ObjectCacheAppCache(),
+            new DictionaryAppCache(),
+            new IsolatedCaches(t => new ObjectCacheAppCache()));
+
+        var provider = ScopeProvider;
+        var scopeAccessor = ScopeAccessor;
+
+        using (var scope = provider.CreateScope())
+        {
+            var repository = CreateRepository((IScopeAccessor)provider, out var contentTypeRepository, realCache);
+
+            var udb = scopeAccessor.AmbientScope.Database;
```

Comment:
Just a pet pieve of mine, I don't like the "udb" naming 🤣 
I would much rather it be `database` or if you must `umbracoDatabase` 🙈
(This named variable is in all the new tests, so I can't even make it easy with suggestions, sorry)